colossalai/kernel/cuda_native/flash_attention.py

```diff
@@ -146,7 +157,7 @@ def forward(self,
 
             out = memory_efficient_attention(query, key, value, attn_bias=attn_bias, p=self.dropout, scale=self.scale)
 
-            if attn_mask_type == AttnMaskType.padding and batch_size > 1:
+            if attn_mask_type in [AttnMaskType.padding, AttnMaskType.paddedcausal] and batch_size > 1:
```

Comment:
How about this way to check if padded: `attn_mask_type.value % 2 == 1`

Reply:
> How about this way to check if padded: `attn_mask_type.value % 2 == 1`

Updated.